src/signing_key.rs

```diff
@@ -106,6 +115,49 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        if pki.algorithm == ALGORITHM_ID {
+            SigningKey::try_from(pki.private_key)
+        } else {
+            Err(Error::MalformedSecretKey)
+        }
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> PrivateKeyDocument {
+        let pki = PrivateKeyInfo {
+            algorithm: ALGORITHM_ID,
+            private_key: &self.seed,
+        };
+        PrivateKeyDocument::try_from(pki).unwrap()
```

Comment:
There's a [`From` impl available here](https://docs.rs/pkcs8/0.6.0/pkcs8/struct.PrivateKeyDocument.html#impl-From%3CPrivateKeyInfo%3C%27_%3E%3E):

```suggestion
        PrivateKeyInfo {
            algorithm: ALGORITHM_ID,
            private_key: &self.seed,
        }.into()
```